includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -0,0 +1,301 @@
+<?php
+/**
+ * Class UPE_Payment_Gateway
+ *
+ * @package WCPay\Payment_Methods
+ */
+
+namespace WCPay\Payment_Methods;
+
+use WCPay\Logger;
+use WC_Payment_Gateway_WCPay;
+use WC_Payments_Account;
+use WC_Payments_Action_Scheduler_Service;
+use WC_Payments_API_Client;
+use WC_Payments_Customer_Service;
+use WC_Payments_Token_Service;
+use WC_Payments;
+use WC_Payments_Utils;
+use Exception;
+use WCPay\Exceptions\Process_Payment_Exception;
+
+
+/**
+ * UPE Payment method extended from WCPay generic Gateway.
+ */
+class UPE_Payment_Gateway extends WC_Payment_Gateway_WCPay {
+	/**
+	 * Internal ID of the payment gateway.
+	 *
+	 * @type string
+	 */
+	const GATEWAY_ID = 'woocommerce_payments';
+
+	const METHOD_ENABLED_KEY = 'enabled';
+
+	/**
+	 * UPE Constructor same parameters as WC_Payment_Gateway_WCPay constructor.
+	 *
+	 * @param WC_Payments_API_Client               $payments_api_client      - WooCommerce Payments API client.
+	 * @param WC_Payments_Account                  $account                  - Account class instance.
+	 * @param WC_Payments_Customer_Service         $customer_service         - Customer class instance.
+	 * @param WC_Payments_Token_Service            $token_service            - Token class instance.
+	 * @param WC_Payments_Action_Scheduler_Service $action_scheduler_service - Action Scheduler service instance.
+	 */
+	public function __construct( WC_Payments_API_Client $payments_api_client, WC_Payments_Account $account, WC_Payments_Customer_Service $customer_service, WC_Payments_Token_Service $token_service, WC_Payments_Action_Scheduler_Service $action_scheduler_service ) {
+		parent::__construct( $payments_api_client, $account, $customer_service, $token_service, $action_scheduler_service );
+		$this->method_title       = __( 'WooCommerce Payments - UPE', 'woocommerce-payments' );
+		$this->method_description = __( 'Accept payments via Stripe.', 'woocommerce-payments' );
+		$this->title              = __( 'WooCommerce Payments', 'woocommerce-payments' );
+		$this->description        = __( 'You will be redirected to Stripe.', 'woocommerce-payments' );
+
+		add_action( 'wp', [ $this, 'maybe_process_redirect_order' ] );
+	}
+
+	/**
+	 * Creates payment intent using current cart and store details.
+	 *
+	 * @return array
+	 */
+	public function create_payment_intent() {
+		$amount         = WC()->cart->get_cart_contents_total();
```

Comment:
Thanks for this suggestion: added `false` as a context parameter to `get_total`, so it returns the amount instead of HTML, but this is corrected in my last commit.